python/tk_multi_publish2/base_hooks/collector.py

```diff
@@ -155,10 +166,224 @@ def update_value(self):
                 # Only overwrite valid values
                 item.properties[self._name] = self._value
 
+            # Emit that our value has changed, after updating the values
+            # otherwise other widgets still get the old properties value
+            self.value_changed.emit()
+
         @property
         def items(self):
             return self._items
 
+    class FieldsPropertyWidget(PluginBase.ValueWidgetBaseClass):
```

Comment:
This should inherit from `PropertyWidget`

Reply:
You can follow the paradigm for `SettingWidget` and abstract functionality into a `PropertyWidgetBaseClass`.

Reply:
If you look at the relationship between `SettingWidgetBaseClass` `SettingWidget` and `TemplateSettingWidget`, you can see how my suggestion would work. You can of course override `__init__()` on the derived `FieldsPropertyWidget` subclass so that it does not instantiate `value_widget` in the same way, but 1) there is still a lot of shared functionality between the two that we should avoid duplicating and 2) its cleaner design that all property widget subclasses derive from a common property widget interface class.